utils/python/CIME/case_run.py

```diff
@@ -182,35 +183,46 @@ def resubmit_check(case):
         submit(case, job=job, resubmit=True)
 
 ###############################################################################
-def do_data_assimilation(da_script, lid):
+def do_data_assimilation(da_script, cycle, data_assimilation_cycles, lid):
 ###############################################################################
-    cmd = da_script + "1> da.log.%s 2>&1" %(lid)
+    cmd = da_script + " 1> da.log.%s %d %d 2>&1" %(lid, cycle, data_assimilation_cycles)
     logger.debug("running %s" %da_script)
     run_cmd_no_fail(cmd)
     # disposeLog(case, 'da', lid)  THIS IS UNDEFINED!
 
 ###############################################################################
+def new_lid():
+###############################################################################
+    lid = time.strftime("%y%m%d-%H%M%S")
+    os.environ["LID"] = lid
+    return lid
+
```

Comment:
Need to assign a new LID after every data assimilation step to avoid log files being overwritten.